Sources/Afluent/SequenceOperators/OutputSequence.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  OutputSequence.swift
+//  Afluent
+//
+//  Created by Roman Temchenko on 2025-03-05.
+//
+
+import Foundation
+
+extension AsyncSequences {
+    public struct OutputAt<Upstream: AsyncSequence & Sendable>: AsyncSequence, Sendable {
+        public typealias Element = Upstream.Element
+        let upstream: Upstream
+        let index: Int
+
+        public struct AsyncIterator: AsyncIteratorProtocol {
+            var upstreamIterator: Upstream.AsyncIterator
+            let index: Int
+            var finished = false
```

Comment:
What's the purpose of this variable? I commented it out and ran the tests and they still passed. 

If it's about this runtime contract: https://developer.apple.com/documentation/swift/asynciteratorprotocol#End-of-Iteration we have a couple choices I think:
- Just rely on the upstream to do this (no need to track state in this operator)
- Don't rely on the upstream

In either case, we should probably write a test for it.

Reply:
I followed `collect` implementation.
But I likely don't need it and can simply check for `currentIndex` being larger than `index`.

Reply:
Removed `finished` var.